src/Zynga/Framework/PgData/V1/PgModel/Writer.hh

```diff
@@ -100,12 +98,12 @@ class Writer implements WriterInterface {
       $pgCache = $pgModel->cache();
       $dataCache = $pgCache->getDataCache();
 
-      if ($dataCache->isLocked($obj) === false) {
-        throw new Exception(
-          'No lock acquired before calling save on obj='.
-          $obj->export()->asJSON(),
-        );
-      }
```

Comment:
I thought I would talk with you  before taking this out. 
Compare the Write->add method with save & delete. 
Add does it nicely:
$locked = $pgCache->lockRowCache($row);
 if ($locked === true) {
  //do all the locking and rest of the stuff
}

I think we should apply same for save & delete and we just allow user to set $shouldUnlock flag and let framework do the locking & unlocking stuff. what's your thoughts?